processing/src/main/java/io/druid/query/groupby/GroupByQueryHelper.java

```diff
@@ -94,8 +95,7 @@ public String apply(DimensionSpec input)
           false,
           true,
           sortResults,
-          Math.min(query.getContextValue(CTX_KEY_MAX_RESULTS, config.getMaxResults()), config.getMaxResults()),
-          bufferPool
+          Math.min(query.getContextValue(CTX_KEY_MAX_OFF_HEAP_SIZE, config.getMaxResults()), config.getMaxOffheapSize())
```

Comment:
copy/paste error, fixed, it is only supposed to rely on max off-heap limit and not on maxRows.